src/oop/object_independence/hard/q2/MainTest.java

```diff
@@ -0,0 +1,237 @@
+        Object[][] arguments = {
+                {c, String.class},
+                {g, String.class},
+                {a, int.class}
+        };
+        String incorrectVisibilityModifier = "Your %s constructor has the incorrect visibility modifier.";
+        assertTrue(classInstance.hasModifier(new Class[]{String.class, String.class, int.class}, "public"),
+                String.format(incorrectVisibilityModifier, objectClassName));
+        Object instance = classInstance.createInstance(arguments);
+        String incorrectAttributeValue =
+                "Your %s constructor does not correctly initialize the %s attribute based on the passed parameters.";
+        _assertEquals(c, classInstance.getFieldValue(instance, stringAttributeName1),
+                String.format(incorrectAttributeValue, objectClassName, stringAttributeName1));
+        _assertEquals(g, classInstance.getFieldValue(instance, stringAttributeName2),
+                String.format(incorrectAttributeValue, objectClassName, stringAttributeName2));
+        _assertEquals(a, classInstance.getFieldValue(instance, intAttributeName1),
+                String.format(incorrectAttributeValue, objectClassName, intAttributeName1));
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void fishHasCorrectGetMethods(String colour, String gender, int age) throws Throwable {
+        Object[][] arguments = {
+                {colour, String.class},
+                {gender, String.class},
+                {age, int.class}
+        };
+        Object instance = classInstance.createInstance(arguments);
+        String incorrectGetMethods = "Your %s method does not return the value of the %s attribute.";
+
+        Object output = classInstance.callMethod(getStringAttributeName1, new String[]{"public"}, instance);
+        _assertEquals(colour, output, String.format(incorrectGetMethods, getStringAttributeName1, stringAttributeName1));
+        output = classInstance.callMethod(getStringAttributeName2, new String[]{"public"}, instance);
+        _assertEquals(gender, output, String.format(incorrectGetMethods, getStringAttributeName2, stringAttributeName2));
+        output = classInstance.callMethod(getIntAttributeName1, new String[]{"public"}, instance);
+        _assertEquals(age, output, String.format(incorrectGetMethods, getIntAttributeName1, intAttributeName1));
+    }
+
+    private static Stream<Arguments> setMethodsInputProvider() {
+        return Stream.of(
+                Arguments.of("Red", "Female", 18, "Pansy", "NA", 28),
+                Arguments.of("Pink", "Xeno", 2, "Brown", "Male", 198),
+                Arguments.of("Cosmic Latte", "Male", 0, "Scarlet", "Female", 32),
+                Arguments.of("Amethyst", "NA", 16, "Purple", "Male", 2),
+                Arguments.of("Orange", "Male", 678, "Yellow", "NA", 0)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("setMethodsInputProvider")
```

Comment:
We should make different test for each method here both for accessors and mutators. Because if they write one of the corectly they should get the test for that.